### PR TITLE
Message Loader (New Feature)

### PR DESCRIPTION
**Pull request by Dorian30**
## Summary

- Made a message loader animation between responses through a new dispatcher `toggleMsgLoader`
- Implemented an example of how to use it 

## Screenshots 

rw

### CODE REVIEW

File: src/components/Widget/components/Conversation/components/Messages/styles.scss [hunk 1]
```diff
@@ -21,3 +21,51 @@
...
+  margin: 10px;
+  display: none;
+
+  &.active {
+    display: flex;
+  }
+}
+
+.loader-container {
+  background-color: #f4f7f9;
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Messages/styles.scss]: Add color variable to the `colors` file

File: src/components/Widget/components/Conversation/components/Messages/styles.scss [hunk 2]
```diff
@@ -21,3 +21,51 @@
...
+  max-width: 215px;
+  text-align: left;
+}
+
+.loader-dots {
+  display: inline-block;
+  height: 4px;
+  width: 4px;
+  border-radius: 50%;
+  background: gray;
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Messages/styles.scss]: Use variable for the color

File: src/components/Widget/components/Conversation/components/Messages/index.js [hunk 3]
```diff
@@ -50,6 +50,13 @@ class Messages extends Component {
             {this.getComponentToRender(message)}
           </div>
         )}
+        <div className={`loader ${typping && 'active'}`}>
```
Comment (mcallegari10) [inline src/components/Widget/components/Conversation/components/Messages/index.js]: I think the loader should be a component of it's own

File: src/components/Widget/components/Conversation/components/Messages/styles.scss (hunk 2 above)
Comment (Dorian30) [reply src/components/Widget/components/Conversation/components/Messages/styles.scss]: On it !

File: src/components/Widget/components/Conversation/components/Messages/components/Loader/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+<div className={`loader ${props.typping && 'active'}`}>
+  <div className="loader-container">
+    <span className="loader-dots"></span>
+    <span className="loader-dots"></span>
+    <span className="loader-dots"></span>
+  </div>
+</div>;
+
+Loader.propTypes = {
+  typping: PropTypes.bool,
```
Comment (GainorB) [inline src/components/Widget/components/Conversation/components/Messages/components/Loader/index.js]: Misspell. Should be `typing`.

File: src/components/Widget/components/Conversation/components/Messages/index.js [hunk 5]
```diff
@@ -38,7 +39,7 @@ class Messages extends Component {
   }
 
   render() {
-    const { messages, profileAvatar } = this.props;
+    const { messages, profileAvatar, typping } = this.props;
```
Comment (GainorB) [inline src/components/Widget/components/Conversation/components/Messages/index.js]: Misspell. Should be `typing`.

File: src/components/Widget/components/Conversation/components/Messages/index.js [hunk 6]
```diff
@@ -50,6 +51,7 @@ class Messages extends Component {
             {this.getComponentToRender(message)}
           </div>
         )}
+        <Loader typping={typping} />
```
Comment (GainorB) [inline src/components/Widget/components/Conversation/components/Messages/index.js]: Misspell. Should be `typing`.

File: src/components/Widget/components/Conversation/components/Messages/index.js [hunk 7]
```diff
@@ -61,5 +63,6 @@ Messages.propTypes = {
 };
 
 export default connect(store => ({
-  messages: store.messages
+  messages: store.messages,
+  typping: store.behavior.get('msgLoader')
```
Comment (GainorB) [inline src/components/Widget/components/Conversation/components/Messages/index.js]: Misspell. Should be `typing`.

**Comment by brandonsturgeon** [comment]
Does this just wait 2 seconds before sending the message?

**Pull request merged**